volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -1761,3 +1764,132 @@ def get_capabilities(self) -> int:
             )
 
         return cap_value & self.get_kernel_cap_full()
+
+
+class timespec64(objects.StructType):
+    def to_datetime(self) -> datetime:
+        """Returns the respective aware datetime"""
+
+        dt = renderers.conversion.unixtime_to_datetime(self.tv_sec + self.tv_nsec / 1e9)
+        return dt
+
+
+class inode(objects.StructType):
+    def is_valid(self) -> bool:
+        # i_count is a 'signed' counter (atomic_t). Smear, or essentially a wrong inode
+        # pointer, will easily cause an integer overflow here.
+        return self.i_ino > 0 and self.i_count.counter >= 0
+
+    def is_dir(self) -> bool:
+        """Returns True if the inode is a directory"""
+        return stat.S_ISDIR(self.i_mode) != 0
+
+    def is_reg(self) -> bool:
+        """Returns True if the inode is a regular file"""
+        return stat.S_ISREG(self.i_mode) != 0
+
+    def is_link(self) -> bool:
+        """Returns True if the inode is a symlink"""
+        return stat.S_ISLNK(self.i_mode) != 0
+
+    def is_fifo(self) -> bool:
+        """Returns True if the inode is a FIFO"""
+        return stat.S_ISFIFO(self.i_mode) != 0
+
+    def is_sock(self) -> bool:
+        """Returns True if the inode is a socket"""
+        return stat.S_ISSOCK(self.i_mode) != 0
+
+    def is_block(self) -> bool:
+        """Returns True if the inode is a block device"""
+        return stat.S_ISBLK(self.i_mode) != 0
+
+    def is_char(self) -> bool:
+        """Returns True if the inode is a char device"""
+        return stat.S_ISCHR(self.i_mode) != 0
+
+    def is_sticky(self) -> bool:
+        """Returns True if the sticky bit is set"""
+        return (self.i_mode & stat.S_ISVTX) != 0
+
+    def get_inode_type(self) -> str:
+        """Returns inode type name
+
+        Returns:
+            The inode type name
+        """
+        if self.is_dir():
+            return "DIR"
+        elif self.is_reg():
+            return "REG"
+        elif self.is_link():
+            return "LNK"
+        elif self.is_fifo():
+            return "FIFO"
+        elif self.is_sock():
+            return "SOCK"
+        elif self.is_char():
+            return "CHR"
+        elif self.is_block():
+            return "BLK"
+        else:
+            return renderers.UnparsableValue()
+
+    def get_inode_number(self) -> int:
+        """Returns the inode number"""
+        return int(self.i_ino)
+
+    def ___time_member_to_datetime(self, member) -> datetime:
```

Comment:
`__` at the start has a specific meaning within python.  Unlike just `_` which informally means private, but actually has no effect, `__` I think prevents inheritance of the class and does alter the way python works.

I don't know if this is to mimic a linux function name and I'm all for making the function names as similar to linux as possible, but when they conflict with the way python functions I don't think we can keep them in sync.  If this isn't supposed to be called outside the class it should only have one `_` (so the class could still be inherited by someone and function) and if it is supposed to be called outside the class, then ideally it wouldn't start with any `_`.

Reply:
Yeah, exactly. It's a private function to prevent inheritance and outside use. It's just an internal helper for other methods, so it doesn't make sense to inherit or use it directly from outside. May I leave this like that or still need to be converted to _ ?

Reply:
I'd prefer it as a `_` because if someone wants to extend this class, they will need that function to be inheritted, so fully private makes that much more difficult.

Reply:
done